compiler/llvm/clangUtil.cpp

```diff
@@ -2253,12 +2271,18 @@ void runClang(const char* just_parse_filename) {
     printf("\n");
   }
 
+  bool parseOnly = (just_parse_filename != NULL);
+
   // Initialize gGenInfo
   // Toggle LLVM code generation in our clang run;
   // turn it off if we just wanted to parse some C.
-  gGenInfo = new GenInfo();
-
-  bool parseOnly = (just_parse_filename != NULL);
+  if (parseOnly) {
+    gGenInfo = new GenInfo();
```

Comment:
IMO it would be nice to always initialize gGenInfo outside of this function, but I understand that might not be super easy.